docs/DeveloperGuide.md

```diff
@@ -176,6 +177,41 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### \[Proposed\] View student/lesson feature
+
+#### Proposed Implementation
+
+The proposed view student/lesson mechanism is facilitated by `ModelManager`. It implements `Model`, stored internally as a `modelManagerStateList` and `currentStatePointer`. Additionally, it implements the following operations:
+
+* `ModelManager#viewStudent()` — Updates student panel with the student of interest and lesson panel with the lessons the student of interest is in.
+* `ModelManager#viewLesson()` — Updates lesson panel with the lesson of interest and student panel with the students that are in this lesson of interest.
+
+This operation is exposed in the `Model` interface as `Model#viewStudent()` and `Model#viewLesson()`.
+
+Given below is an example usage scenario and how the view student mechanism behaves at each step.
+
+Step 1. The user launches the application for the first time. The `ModelManager` will be initialized with the initial model manager state, and the `currentStatePointer` pointing to that single model manager state.
+
+![ViewStudentState0](images/ViewStudentState0.png)
+
+Step 2. The user executes `view -s 1` command to view the 1st student in TutorAid. The `view` command calls `Model#viewStudent()`, causing the modified state of model manager after the `view -s 1` command executes to be saved in the `modelManagerStateList`, and the `currentStatePointer` pointing to that model manager state.
```

Comment:
Should `Model#viewStudent()` be called by the `view -s` command instead?

Reply:
Right yes I have updated it thank you.